### PR TITLE
Try to close dask threads after usage.

### PR DESCRIPTION
**Pull request by Carreau**
Try to close dask threads after usage.

This tries to make sure that dask thread worker is closed after each test.
This is not in it's own an issue, but from some of the mac failures I
have the impression there there are thread problems (one of the recent
failure was in IPython's history saving thread), so one of the things
I'd like to enable in pytest fixture is that the number of thread
before/after tests has not changed, unless the test if flagged
specifically.

This ensure that the above is true – at least from the dask scheduler.
We make sure the threaded scheduler is not setup.

Later, once other threads have been cleaned up as well,
we can make sure that the number of thread before/after is unchanged.

In addition this also make IPython history saving in-memory to save a
thread on having to write to disk, and put a reminder that widgets leaks
detection has strong false positive if some other errors happen, and I
caught myself forgetting this several times.

### CODE REVIEW

**Comment by Carreau** [comment]
Converting to draft, while this work as is, there is an ordering issue if I try to make sure that the number of threads stays 1 in make napari viewer.

**Comment by Carreau** [comment]
@tlambert03 any idea why this would suddenly cause failure in magicgui related code ?

**Comment by tlambert03** [comment]
no, very strange.  does it repeat locally? or only on CI?

**Comment by Carreau** [comment]
Only CI, there is this message that xvfb can't open display, which may be related.

**Comment by tlambert03** [comment]
the only thing i can think of is perhaps [these few lines in shutdown](https://github.com/python/cpython/blob/4d6c0c0cce05befa06e0cad7351b1303ac048277/Lib/concurrent/futures/thread.py#L233-L235), that join all threads.

It's conceivable that the `ThreadPoolExecutor` is hanging on shutdown, and it might just be a coincidence that the next `qtbot.waitSignal` call happens to be in a magicgui test?  Really not sure though.  maybe try `shutdown(wait=False)`?

**Comment by tlambert03** [comment]
this is still relevant even with mac tests passing right?

**Comment by Carreau** [comment]
> this is still relevant even with mac tests passing right?

I don't know. At least if dask properly closes threads, then I can try to hunt other leaking threads.
My hope is to have a fixture that assert the number of threads don't change between before/after the tests.

Plus the mac tests still take 15+ seconds each for screenshot, and I'm unsure why, they should be much faster.

**Review by tlambert03 — APPROVED** [review]
welp, I'll approve and leave it to you if want to merge :)

**Comment by Carreau** [comment]
Let's merge then.

**Pull request merged**
